src/common.hpp

```diff
@@ -0,0 +1,10 @@
+#include <optional>
+
+#include <fmt/format.h>
+#include <tl/expected.hpp>
+
+#define Some std::make_optional
+#define None std::nullopt
+
+#define Ok(x) (x)
+#define Err tl::make_unexpected
```

Comment:
For those who don't catch up with the current C++ standards, std::format and std::expected are all ratified but are not available in the libcxx/libstdc++ on your platform yet. We can use third party implementations from which the standard library steals.

All in all, these libraries solve compatibility issues you may come across on your platform.